src/index.js

```diff
@@ -145,7 +145,46 @@ const rules = {
         ...optionGroup,
       }));;
     },
-  }
+  },
+  'required-fields': {
+    meta: {
+      schema: {
+        type: 'array',
+        minLength: 1,
+        items: {
+          additionalProperties: false,
+          properties: {
+            ...defaultRuleProperties,
+            requiredFields: {
+              type: 'array',
+              items: {
+                type: 'string',
+              },
+            },
+          },
+          oneOf: [
+            {
+              required: ['schemaJson'],
+              not: { required: ['schemaJsonFilepath'] },
+            },
+            {
+              required: ['schemaJsonFilepath'],
+              not: { required: ['schemaJson'] },
+            },
+          ],
+        },
+      },
+    },
+    create: context => {
+      return createRule(context, optionGroup =>
```

Comment:
`context.options` [has the options you want](http://eslint.org/docs/developer-guide/working-with-rules-new#contextoptions)